ui/app/src/Views/SilenceList/SilenceView.elm

```diff
@@ -26,15 +29,20 @@ view showConfirmationDialog silence =
         , class "align-items-start list-group-item border-0 p-0 mb-4"
         ]
         [ div [ class "w-100 mb-2 d-flex align-items-start" ]
-            [ case silence.status.state of
-                Active ->
-                    dateView "Ends" silence.endsAt
+            [ case silence.status of
+                Just status ->
+                    case status.state of
+                        Active ->
+                            dateView "Ends" silence.endsAt
 
-                Pending ->
-                    dateView "Starts" silence.startsAt
+                        Pending ->
+                            dateView "Starts" silence.startsAt
 
-                Expired ->
-                    dateView "Expired" silence.endsAt
+                        Expired ->
+                            dateView "Expired" silence.endsAt
+
+                Nothing ->
+                    text ""
```

Comment:
Can we get rid of Maybe here or flatten this possibility into the status union type so that it can be Active, Pending, Expired or Undefined?

Reply:
The definition of `Silence` is generated based on the new OpenAPI specification of the API v2. I would prefer not to have an additional internal type which converts to the generated one.

Why does the specification not enforce the `status` property: E.g. when creating a silence that status property is not yet known. Thereby the API does not enforce the property.

Is that enough context? What are your thoughts?